src/utils/async.ts

```diff
@@ -48,3 +50,43 @@ export async function asyncIterableToArray<T>(
   const iter = source[Symbol.asyncIterator]();
   return asyncIteratorToArray(iter);
 }
+
+export function mapEitherAsyncIterator<E, T, V>(
+  iter: AsyncIterator<Either<E, T>, Either<E, T>>,
```

Comment:
this looks too specific, I would implement a `filterAsyncIterator` instead:

```ts
filterAsyncIterator<T>(
  iter: AsyncIterator<T>,
  predicate: (t: T) => boolean
): AsyncIterator<T>;
```

then `mapEitherAsyncIterator` becomes:

```ts
filterAsyncIterator(mapAsyncIterator(iter, _ => _.isRight() ? _.value : undefined), isDefined);
```

where `isDefined` is a guard

Reply:
Fixed in 8f4c631